cs/src/Connections/RefreshingTunnelEventArgs.cs

```diff
@@ -0,0 +1,66 @@
+// <copyright file="RefreshingTunnelEventArgs.cs" company="Microsoft">
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// </copyright>
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DevTunnels.Contracts;
+using Microsoft.DevTunnels.Management;
+
+namespace Microsoft.DevTunnels.Connections;
+
+/// <summary>
+/// Event args for tunnel refresh.
+/// </summary>
+public class RefreshingTunnelEventArgs : EventArgs
+{
+    /// <summary>
+    /// Create a new instance of <see cref="RefreshingTunnelEventArgs"/>.
+    /// </summary>
+    public RefreshingTunnelEventArgs(
+        string tunnelAccessScope,
+        Tunnel? tunnel,
+        ITunnelManagementClient? managementClient,
+        bool includePorts,
+        CancellationToken cancellation)
+    {
+        TunnelAccessScope = Requires.NotNull(tunnelAccessScope, nameof(tunnelAccessScope));
+        Tunnel = tunnel;
+        ManagementClient = managementClient;
+        IncludePorts = includePorts;
+        Cancellation = cancellation;
+    }
+
+    /// <summary>
+    /// Get tunnel access scope for this tunnel client or host.
+    /// </summary>
+    public string TunnelAccessScope { get; }
+
+    /// <summary>
+    /// Get the tunnel being refreshed.
+    /// </summary>
+    public Tunnel? Tunnel { get; }
```

Comment:
Why is this nullable, won't there always be a tunnel being refreshed

Reply:
In some cases relay clients may connect to Relay without the `Tunnel` object. This is used for example by Relay service itself. The tunnel refreshing may kick off without a known tunnel object.